### DISCUSSION TITLE
beginner here - having trouble pushing commits to repository

### DISCUSSION THREAD

**Discussion by edgarflopez** [community/community#28601] [New to GitHub] [answered]
I get this error: 'fatal: remote error is not a valid repository' And then when I go to github I cant see the files I am pushing to my repository.  but when I look in the file directory they are there.

edgar@LAPTOP-Q46CRTGI MINGW64 ~/onedrive/desktop/projects/run-buddy (main)
$ git remote -v
origin  git@github.com:edgarflopez/run-buddy.gitgit@github.com:edgarflopez/run-buddy.git (fetch)
origin  git@github.com:edgarflopez/run-buddy.gitgit@github.com:edgarflopez/run-buddy.git (push)

edgar@LAPTOP-Q46CRTGI MINGW64 ~/onedrive/desktop/projects/run-buddy (main)
$ git push origin main
Enter passphrase for key '/c/Users/edgar/.ssh/id_ed25519':
**fatal: remote error:
   is not a valid repository name
  Visit https://support.github.com/ for help**

**Comment by airtower-luna** [comment, marked as answer]
This looks really odd:

> ```
> $ git remote -v
> origin git@github.com:edgarflopez/run-buddy.gitgit@github.com:edgarflopez/run-buddy.git (fetch)
> origin git@github.com:edgarflopez/run-buddy.gitgit@github.com:edgarflopez/run-buddy.git (push)
> ```

It looks like your remote repository URLs got duplicated somehow, maybe a copy&paste error? Whatever the cause, you can fix the URL with [`git remote set-url`](https://git-scm.com/docs/git-remote#Documentation/git-remote.txt-emset-urlem):

```
$ git remote set-url origin git@github.com:edgarflopez/run-buddy.git
$ git remote set-url --push origin git@github.com:edgarflopez/run-buddy.git
```

**Reply by edgarflopez** [reply]
Thank you! That worked! 

At first when I did the git remote commands - nothing happened but then I went and did a git push origin main command again and it worked!
I sure do appreciate that.

**Reply by airtower-luna** [reply]
Glad to hear it worked! And yes, that's how it's supposed to work: The `set-url` commands fixed the URLs, and after that `push` had a valid remote URL to use. Marking the solution would be appreciated. :slightly_smiling_face:

**Reply by afeefuddin** [reply]
@airtower-luna Thanks that was really helpful

**Reply by Ishita-Trivedi** [reply]
Thanks its wonder full!

**Comment by xjf255** [comment]
@airtower-luna Thanks <3

**Comment by Yakutmuder9** [comment]
@airtower-luna  Thanks!
